stock_storage_type/models/stock_location.py

```diff
@@ -301,22 +314,38 @@ def _compute_location_will_contain_lot_ids(self):
         "only_empty",
     )
     def _compute_location_is_empty(self):
-        for rec in self:
-            # No restriction should apply on customer/supplier/...
-            # locations and we don't need to compute is empty
-            # if there is no limit on the location
-            if not rec._should_compute_location_is_empty():
-                # avoid write if not required
-                if not rec.location_is_empty:
-                    rec.location_is_empty = True
-                continue
+        # No restriction should apply on customer/supplier/...
+        # locations and we don't need to compute is empty
+        # if there is no limit on the location
+        only_empty_locations = self.filtered(
+            lambda l: not l._should_compute_location_is_empty()
+        )
+        only_empty_locations.update({"location_is_empty": True})
```

Comment:
```suggestion
        only_empty_locations.filter(lambda l: not l.location_is_empty).update({"location_is_empty": True})
```
?

Reply:
In fact, the impact is really not significant